messaging/broker.go

```diff
@@ -818,6 +878,80 @@ func (t *Topic) WriteMessage(m *Message) error {
 	return nil
 }
 
+// Truncate attempts to delete topic segments such that the total size of the topic on-disk
+// is equal to or less-than maxSize. Returns the number of bytes deleted, and error if any.
+// This function is not guaranteed to be performant.
+func (t *Topic) Truncate(maxSize int64) (nBytesDeleted int64, err error) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+
+	var segments Segments
+	segments, err = ReadSegments(t.Path())
```

Comment:
As long as it also doesn't declare `err` as new. I was concerned it would -- I am using named returns.